ArgsParsing/TypeParsers/IntParser.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ArgsParsing.TypeParsers
+{
+    /// <summary>
+    /// A parser capable of parsing numbers.
+    /// </summary>
+    public class IntParser : BaseArgumentParser<int>
+    {
+        private readonly int _minValue;
+        private readonly int _maxValue;
+
+        public IntParser(int minValue = 0, int maxValue = int.MaxValue)
```

Comment:
you mean minimum 1?

Reply:
int.MinValue

Reply:
oh I read that as _un_signed. I made it default to 0 by default to avoid negative bet bugs and similar. I figure the usecases for passing actual negative numbers is slim

Reply:
tbh it might be better to have a separate unsigned int parser and use unsigned int instead

Reply:
I wanted to specify one type per kind of number, like the old core does it.
For example "PositivePokeyen" would then get an int parser with minValue=1

for that there's no need for multiple different parsers.

Reply:
maybe so, but when I want to parse just an int, I usually expect to be able to recieve negative numbers, if I want positive I would use `unsigned int`, but there's no parser for it